portfolio/src/main/webapp/script.js

```diff
@@ -195,13 +261,81 @@ function createCommentElement(comment) {
   return commentElement;
 }
 
+/**
+ * Disables some of the pagination buttons if the current page
+ * is the first or the last one.
+ */
+function disablePaginationButtons() {
+  let disable = false;
+  if (getCurrentPage() === 1) {
+    // disables first and previous page button
+    disable = true;
+  }
+  document.getElementById("first-page").disabled = disable;
+  document.getElementById("prev-page").disabled = disable;
+  
+  disable = false;
+  if (getCurrentPage() === getTotalPage()) {
+    // disables last and next page button
+    disable = true;
+  }
+  document.getElementById("last-page").disabled = disable;
+  document.getElementById("next-page").disabled = disable;
+}
+
+/**
+ * Returns the current number of the comments page.
+ */
+function getCurrentPage() {
+  const currentPage = document.getElementById("current-page").textContent;
+  return toInt(currentPage, 0);
```

Comment:
Shouldn't default value be 1?

Reply:
No, I use it to recognise negative values: if there is a negative value in the `current-page`, when I click next, it will return 0 and then add 1, so it will return to the first page, otherwise if I click previous, then I get 0, subtract 1 and now I have added a control that if the page number is lower than 1 or greater than the total, the current page is set to 1.

Reply:
How could it be possible that negative value would be inside `current-page`? At start it is 1, then (when calling `setPage`) it cannot be lower than 1.
Right now, because this function (`getCurrentPage`) can return 0 for invalid/negative numbers, you need to have a special case inside `getCommentsOffset`. If you change the "contract" so that `getCurrentPage` always return number higher than or equal to 1, you can simplify other places. Think like this: by default, you're showing page number 1. This is what should be displayed at start and whenever there is something bad with input numbers. Just go on as if the current page is 1. So clicking "next" should take you to page 2.

Reply:
Yes, it should never be negative, but I wanted to handle it anyway, in case something goes wrong.
The problem I see if I set the page to 2 when clicking next is that page 2 may not exist, so I will end up in another error state. That's why I would like to always show page 1 in this case.

Reply:
Thanks to the checks inside `setPage` you're safe from setting an invalid page.
But still you need to specially handle the fact that getCurrentPage() may return 0 in some cases. This is easy to forget, and you did already - in `disablePaginationButtons()`, you're only checking 1 and getTotalPage(). See what I mean? The code would be so much simpler if `getCurrentPage` would return at least 1 in every situation. `getCommentsOffset` also would be simplified.

Reply:
It's true, you're right. I will change that. Thanks!